### PR TITLE
make ci faster

### PR DESCRIPTION
**Pull request by ahnitz**
This is a WIP PR to see if we can make the CI faster

Trying so far
* run unittest same time as checking help messages 
* run search / inference tests together
* explicitly run multiple scripts at a time which are needed to generate the docs page

### CODE REVIEW

**Comment by ahnitz** [comment]
@spxiwh I think this moves enough in the right direction to consider at this point. The major changes are noted at the topic. An additional note is that I've updated the scripts to ensure they use single core to make it faster to run a couple in parallel.

File: docs/conf.py
Comment (spxiwh) [inline docs/conf.py:318]: Remove debugging line?

File: docs/conf.py
Comment (spxiwh) [inline docs/conf.py:326]: More debugging lines??

File: docs/conf.py
Comment (spxiwh) [inline docs/conf.py:303]: I guess this is 2 because of the CI runners ... But building docs locally, 2 might not be the best choice. Not sure how to make this choice at a higher level??

File: examples/inference/margtime/run.sh
Comment (spxiwh) [inline examples/inference/margtime/run.sh:3]: We should have one inference example somewhere in the testsuite run for nprocesses > 1.

**Review by spxiwh — APPROVED** [review]
A few minor comments on what's here. But I think this all looks good.

**Comment by spxiwh** [comment]
As a note, I don't think this is going to pass the auto-merge tests because it's expecting tests that are now removed. We might need to admin-merge this when ready.

File: docs/conf.py
Comment (ahnitz) [reply docs/conf.py:303]: Yeah, that's tricky. There's also the problem that memory useage may or may not be conducive to doing something simple like setting it to the number of available threads.

**Comment by ahnitz** [comment]
@spxiwh Yes, that's right. Once this is merged, I'll update the required list and ask people to rebase their PRs.

**Pull request merged**
